cmd/contour/serve.go

```diff
@@ -215,32 +215,35 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	}
 
 	// step 4. register our resource event handler with the k8s informers.
-	var informers []cache.SharedIndexInformer
 
-	informers = registerEventHandler(informers, dynamicInformers.ForResource(ingressroutev1.IngressRouteGVR).Informer(), dynamicHandler)
-	informers = registerEventHandler(informers, dynamicInformers.ForResource(ingressroutev1.TLSCertificateDelegationGVR).Informer(), dynamicHandler)
-	informers = registerEventHandler(informers, dynamicInformers.ForResource(projectcontour.HTTPProxyGVR).Informer(), dynamicHandler)
-	informers = registerEventHandler(informers, dynamicInformers.ForResource(projectcontour.TLSCertificateDelegationGVR).Informer(), dynamicHandler)
-	informers = registerEventHandler(informers, coreInformers.Core().V1().Services().Informer(), eventRecorder)
+	// synclist holds a list of informer HasSynced() functions, which are used later
+	// to wait for all the informers to finish syncing for the first time.
+	synclist := &SyncList{}
```

Comment:
var syncList SyncList

Reply:
I don't think this comment really adds much, maybe called the variable `informerSyncedList` would remove the need for the comment.